jax/interpreters/numpy_eval.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from contextlib import contextmanager
+from typing import Any, Dict, Callable
+
+from jax import config, core
+from jax.interpreters.xla import DeviceArray
+
+np_impl: Dict[core.Primitive, Callable[..., Any]] = {}
+is_tracing = False
+
+class NumpyEvalTrace(core.Trace):
+  def pure(self, x): return x
+  lift = sublift = pure
+
+  def process_primitive(self, primitive, tracers, params):
+    impl = np_impl.get(primitive)
```

Comment:
The numpy_eval_lax.py file feels a bit redundant. What if we delete that file, and just replace this line with something like `getattr(lax_reference, primitive.name, None)`?

Reply:
Have merged the rules into `lax_reference.py`. `getattr(lax_reference, primitive.name, None)` wouldn't work well in general due to mismatch of lax primitives and `lax` method signatures for ~20 out of 100 primitives.